example/bufferCopy/src/bufferCopy.cpp

```diff
@@ -257,3 +257,9 @@ auto main() -> int
 
     return EXIT_SUCCESS;
 }
+
+auto main() -> int
+{
+    // execute the example once for each enabled accelerator
+    return std::apply([](auto const&... tags) { return (example(tags) || ...); }, alpaka::EnabledAccTags{});
```

Comment:
Users will be challenged and might think that this is the way alpaka programs are written. It is an extra layer of difficulty for a novice user. ( I will try to find a solution by using `std::variant `and `std::visit `but seems not simpler than this solution. )

Reply:
Can't we solve this at CI level?

Reply:
Call `ctest -v`  many times in run_test.sh with different arguments? ( Add arguments to example's main function (i.e. argv) in cpp and use  CMAKE_CTEST_ARGUMENTS somehow? )

Reply:
Adding arguments will increase the complexity much more. The problem is we write code which can be executed on many devices so it is not like other projects where a simple main is enough.
This PR solves an issue that we are not able to execute our examples on all devices.
The now-shown CI issues show how easy it is to run the example with all accelerators.

Reply:
> Users will be challenged and might think that this is the way alpaka programs are written. It is an extra layer of difficulty for a novice user.

I fully agree.

If you want to write tests like this, OK... I personally strongly dislike this, but I won't object to obscure unreadable code in the tests, as long as the compiler and CI are happy.

But this should definitely not be how we write an example.